gm4_damaging_stonecutters/data/damaging_stonecutters/functions/damage.mcfunction

```diff
@@ -0,0 +1,14 @@
+#damages entity
+#@s - @e[type=!#damaging_stonecutters:stonecutter_immune]
+#called by damaging_stonecutters:check_damage
+
+#damage entity
+effect give @s minecraft:absorption 1 0 true
+effect give @s[type=!#damaging_stonecutters:undead] minecraft:instant_damage 1 0 true
```

Comment:
You could use the wither effect to avoid needing to check if the mob is undead.

Reply:
We *did* use the wither effect at first, however we found too many downsides and just copied the code from heart canisters